### PR TITLE
Add support for RNG passthrough

### PR DESCRIPTION
**Pull request by rmohr**
Allow passing through /dev/random from the host to the VM. This speeds
up pretty much everything which needs entropy inside the VM.

### CODE REVIEW

**Comment by infernix** [comment]
A few typos, but LGTM. @pronix ?

File: lib/vagrant-libvirt/config.rb [hunk 1]
```diff
@@ -317,6 +323,18 @@ def channel(options={})
         })
       end
 
+      def random(options={})
+        if !options[:model].nil? && options[:model] != "random"
+          raise 'The only supported rng backend is "random".'
+        end
+
+        if @rng == UNSET_VALUE
```
Comment (pronix) [inline lib/vagrant-libvirt/config.rb]: it should be first line of method

Comment (pronix) [inline lib/vagrant-libvirt/config.rb]: and function is not used.

Comment (rmohr) [inline lib/vagrant-libvirt/config.rb]: > and function is not used.

The function is. When specifying the random device in the Vagrantfile like mentioned in the readme:

```
  config.vm.provider :libvirt do |domain|
      domain.random :model => 'random'
  end
```

it is called.

> it should be first line of method

But all other functions have exactly this structure. First input validation, then checking if the target config property is UNSET_VALUE and acting accordingly.

But maybe I miss something here ...

**Comment by rmohr** [comment]
> A few typos, but LGTM. @pronix ?

Fixed the typos where I saw them.

@pronix @infernix thanks for the fast feedback :+1:

File: lib/vagrant-libvirt/action/create_domain.rb [hunk 2]
```diff
@@ -89,6 +89,9 @@ def call(env)
           # USB device passthrough
           @usbs = config.usbs
 
+          # RNG device passthrough
+          @rng =config.rng
```
Comment (infernix) [inline lib/vagrant-libvirt/action/create_domain.rb]: Nitpicking, but... `@rng = config.rng`

:)

**Comment by aspiers** [comment]
Is it just me or is this documented wrong?  It's documented as

    libvirt.random :model => 'random'

but AFAICS from the code it should be

    libvirt.rng :model => 'random'

**Comment by aspiers** [comment]
Nope, it was just me - never mind :-)

**Pull request merged**